tools/install_prerequisites.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+set -eu
+
+# Set the CWD to Omicron's source.
+SOURCE_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+cd "${SOURCE_DIR}/.."
+
+HOST_OS=$(uname -s)
+if [[ "${HOST_OS}" == "Linux" ]]; then
+  sudo apt-get install libpq-dev
+  sudo apt-get install pkg-config
+elif [[ "${HOST_OS}" == "SunOS" ]]; then
+  need=(
+    'pkg:/package/pkg'
+    'build-essential'
+    'library/postgresql-13'
+    'pkg-config'
+    'brand/omicron1/tools'
+  )
+
+  # Perform updates
+  if (( ${#need[@]} > 0 )); then
+    pfexec pkg install -v "${need[@]}" && rc=$? || rc=$?
+    # Return codes:
+    #  0: Normal Success
+    #  4: Failure because we're already up-to-date. Also acceptable.
+    if [ "$rc" -ne 4 ] && [ "$rc" -ne 0 ]; then
```

Comment:
It took me a surprisingly long time to grok what was going on here.  At L24, we're doing `&& rc=$? || rc=$?`.  It might be clearer (and not rely on the reader knowing bash operator precedence) to unconditionally set `rc=$?` after L24.

Tangential: I think [`[[` is generally preferred in bash conditionals](https://stackoverflow.com/questions/669452/are-double-square-brackets-preferable-over-single-square-brackets-in-b).  I don't see a bug here with using single brackets though.

Reply:
Is this more clear?

```bash
rc=0
try_to_do_command || rc=$?
```

The whole point is that I'm trying to check the error condition without failing out of the script, because of `set -e`

Reply:
Thanks -- that's a helpful reminder.  I do think that's clearer.